packages/sc-deployer/src/index.ts

```diff
@@ -23,14 +24,19 @@ const __filename = fileURLToPath(import.meta.url);
 
 const __dirname = path.dirname(__filename);
 
+interface IDeploymentInfo {
+  opId: string;
+  events?: IEvent[];
+}
+
 async function deploySC(
   publicApi: string,
   account: IAccount,
   contracts: ISCData[],
   fee: number = 0,
   maxGas: number = 1_000_000,
   wait: boolean = false,
-) {
+): Promise<IDeploymentInfo> {
```

Comment:
why does it say it returns a promise whereas the return is a js object?

Reply:
async function always returns a Promise

Reply:
And using `await` on it will unwrap the object from the Promise

Reply:
correct!

Reply:
awesome thanks!